### ISSUE TITLE
Can someone help me updating the pyfaidx recipe?

### ISSUE THREAD

**Issue by membranepotential** [bioconda/bioconda-recipes#4985]
Hi!

I saw that pyfaidx was outdated and thought it would be easy to contribute by adding the new version as recipe. Unfortunately it doesn't work.

I used the old recipe as template and bumped to the new version. Additionally I added `noarch: python`. 

`conda-build` failed for both the new version and the older existing version on my computer (macos), complaining that downloads via setuptools are forbidden. It worked though when doing `conda-build --channel defaults`. I thought to give it a shot and pushed to my fork. Travis-CI then failed for Linux but worked for mac.

To solve the initial problem, I found an ugly hack to tell setuptools not to download dependencies. So I put 
`$PYTHON setup.py install --single-version-externally-managed --record=/tmp/record.txt` in the build.sh.

That made it work on my computer. However, TravisCI is still failing with
```
BIOCONDA ERROR BUILD FAILED: the built package /anaconda/conda-bld/noarch/pyfaidx-0.4.9.2-py_0.tar.bz2 cannot be found
BIOCONDA ERROR BUILD SUMMARY: of 1 recipes, 1 failed and 0 were skipped. Details of recipes and environments follow.
BIOCONDA ERROR BUILD SUMMARY: FAILED recipe pyfaidx-0.4.9.2-py_0.tar.bz2, environment CONDA_PERL=5.22.0;CONDA_GMP=5.1;CONDA_NPY=112;CONDA_HTSLIB=1.4;CONDA_PY=27;CONDA_R=3.3.2;CONDA_NCURSES=5.9;CONDA_BOOST=1.61;CONDA_HDF5=1.8.17;CONDA_GSL=1.16
```

I also tried to remove `noarch: python`, but then the linter complains.

The Travis logs can be found here: https://travis-ci.org/membranepotential/bioconda-recipes

My fork with the changes can be found here: https://github.com/membranepotential/bioconda-recipes/tree/pyfaidx/recipes/pyfaidx/0.4.9.2

Cheers and thank you for your help!

**Comment by daler** [comment]
@membranepotential you're right, it should be that easy, but we recently added the `noarch` linter and are realizing it's not working well in practice. The linter has since been removed in https://github.com/bioconda/bioconda-utils/pull/138.

Also, since packages of all versions remain available in the bioconda channel, usually we just bump the version in the main recipe. I don't see a main recipe here, only subdirs for older versions. This would probably be a good candidate for adding a main recipe.

Some packages do need setuptools for building, but it's rare to need it at runtime. The conda skeleton as far as I can tell just copies build requirements over to run requirements, so usually you can just remove the `setuptools` dependency from `requirements:run`.

I just created an updated PR addressing this (and, importantly, after merging the bioconda-utils change that disables the noarch linter) at https://github.com/bioconda/bioconda-recipes/pull/4986.

**Comment by daler** [comment]
Fixed in #4987
